lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1224,11 +1224,11 @@ DINode *SPIRVToLLVMDbgTran::transTypeInheritance(const SPIRVExtInst *DebugInst,
   // No Child operand for NonSemantic debug spec
   SPIRVWord Offset = isNonSemanticDebugInfo(DebugInst->getExtSetKind()) ? 1 : 0;
   DIType *Parent =
-      transNonNullDebugType(BM->get<SPIRVExtInst>(Ops[ParentIdx - Offset]));
+      transDebugInst<DIType>(BM->get<SPIRVExtInst>(Ops[ParentIdx - Offset]));
```

Comment:
Should we add a testcase that requires the reversions on lines 1227 and 1231?  

Also, how is "SPIRV unknown type" generated as expected in DebugInfoTypeInheritance.spvasm with these reversions?

Reply:
I have removed special handling of DebugInfoTypeInheritance with DebugInfoNonde. Reason is that LLVM actually can handle absence of parent on its own, just by not emitting the appropriate metadata. It's tested here: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2341/files#diff-60e39e2de73507f9b785b3994221453dc593d76f50be7f3355d2766f5524d2fdR13